### PR TITLE
Improve SPIMquant directory structure

### LINKED ISSUE #12: BlobDog Returns Zero-Sized Array
Error thrown by rule `filter_blobs_betaamyloid`, likely due to blobdog returning array without width if no blobs found:

```
2024-09-24 16:07:03,865 [    INFO] 
ras2vox: [[-0.5 -0.  -0.  -0. ]
 [-0.  -0.5 -0.  -0. ]
 [-0.  -0.  -0.5 -0. ]
 [ 0.   0.   0.   1. ]]
vox2ras: [[-2.  0. -0. -0.]
 [ 0. -2. -0. -0.]
 [ 0.  0. -2.  0.]
 [ 0.  0.  0.  1.]]
/home/vboxuser/progtools/projects/SPIMquant/spimquant/.snakemake/scripts/tmpusuohgc4.filter_blobs.py:30: RuntimeWarning: Mean of empty slice.
  print(f'points, mean: {points.mean(axis=0)}, max: {points.max(axis=0)}, min: {points.min(axis=0)}')
/home/vboxuser/progtools/venvs/spim/lib/python3.11/site-packages/numpy/core/_methods.py:121: RuntimeWarning: invalid value encountered in divide
  ret = um.true_divide(
Traceback (most recent call last):
  File "/home/vboxuser/progtools/projects/SPIMquant/spimquant/.snakemake/scripts/tmpusuohgc4.filter_blobs.py", line 30, in <module>
    print(f'points, mean: {points.mean(axis=0)}, max: {points.max(axis=0)}, min: {points.min(axis=0)}')
                                                       ^^^^^^^^^^^^^^^^^^
  File "/home/vboxuser/progtools/venvs/spim/lib/python3.11/site-packages/numpy/core/_methods.py", line 41, in _amax
    return umr_maximum(a, axis, None, out, keepdims, initial, where)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: zero-size array to reduction operation maximum which has no identity
```

I think doing a check after the BlobDog output should fix this error? I've seen this one a few times when I was using blobdog function myself.

### PR DESCRIPTION
**Pull request by Karl5766**
Changing SPIMquant directory structure:

1. add .gitignore for config and generated files, now spimquant/config/snakebids.yml is no longer tracked by git
2. instead, a snakebids_template.yml is provided as default config under examples/ folder (currently it needs to be manually copied to spimquant/config/snakebids.yml which is not ideal for command line `spimquant` usage, still need to change the application to automatically use the template when the config file does not exist)
3. move example notebooks into examples/ folder
4. move test empty datasets to Google drive (for now) which are linked in the docs; test ome zarr is now a down-sampled lightsheet image (onuska21) instead of an empty image; some update of the docs
5. fix #12

Looking to fix the issue mentioned in 2) soon.

### CODE REVIEW

**Comment by akhanf** [comment]
We should keep the config where it is, that's where it should live for any snakebids app. 

The current way to run would be clone the repo and edit the config (similar to other snakemake workflows), and we can progressively move to primarily a CLI approach (once we decide on the CLI arguments to include etc), where the user does not have to edit the config files for the majority of use cases.

**Comment by Karl5766** [comment]
Added a commit to copy template over to `config/snakebids.yml` if it is not found. 

I think this way it's easier with git. In the current main branch, if I or someone forking the repo wants to edit the config then `git add -A` will include the local config files which is not desirable. Handling cases like this have been headache for me in the past and git doesn't seem to provide a good solution.

Planning to base the approach around CLI interface would be good as well, but as for now it's not completely replacing it.

File: examples/notebooks/MIP.ipynb
Comment (akhanf) [inline examples/notebooks/MIP.ipynb:1]: At some point I'll try to go through these example notebooks and only keep the relevant ones.. (they are mostly odds and ends right now)

File: spimquant/run.py
[the remark this answers is not in the document]
Comment (Karl5766) [reply spimquant/run.py:1]: It's to make compatible the template snakebids.yml file with the `spimquant` command line usage. 

If not modified, run.py requires a snakebids.yml under `spimquant/config/snakebids.yml` path. Then two choices, one is to keep the template file in there so spimquant automatically use it, another is to keep the template file separate and modify run.py to copy it over if no snakebids.yml if found in default location.

In the first case, the template file is tracked by git and when we adapt our datasets we have to do extra steps during commit and pull to make sure not to accidentally change the template with our own config file specific to the datasets, and this is inconvenient.

In the second case which is implemented in this PR, if copy is not done then users installed the package with `pip install` have to manually go into the `site_packages` folder and make sure the config is copied from template to default location to be used, but this is also inconvenient. Therefore making changes to run.py to automatically copy the file is necessary (if no snakebids.yml file is found run.py will throw an error before parsing Snakefile it seems).

[the remark this answers is not in the document]
Comment (akhanf) [reply spimquant/run.py:1]: I would just rather not deviate considerably from what we do for standard BIDS Apps with snakebids. Yes, until we have a functional CLI we will just have to update the config, but just adding it to the .gitignore ensures that you won't accidentally overwrite..

[the remark this answers is not in the document]
Comment (Karl5766) [reply spimquant/run.py:1]: That's the issue though, adding it to gitignore still causes conflict when pulling, and it becomes confusing as to how to update the template which is ignored by git. git's behavior with adding to .gitignore is sometimes wacky as well that even though it's in .gitignore the index still tracks it if it is not deleted from the index tree (had to admit I don't understand git too well). Having a template is probably a simpler solution.

Is there a reason to not deviate from standard BIDS app? (anything reliant on the implementation of run.py or would be confusing to users of other BIDS app?)

**Comment by Karl5766** [comment]
As a solution after conversation with Ali, config file is not intended to be exposed to user or changed for different datasets. For such cases, using CLI to specify options is preferred. Changed snakebids.yml back to config/ folder.

I'm prepared to merge, @akhanf can you see if now it's good or not?

**Comment by akhanf** [comment]
can you also put back the template files? I see what you were trying to do, but downloading the template is something the workflow should do rather than have the user supply it.. can add the template download functionality in another PR..

**Comment by akhanf** [comment]
also, I make use of the placeholder test directory when doing dry-run tests, we should keep that in the repo too

**Comment by akhanf** [comment]
btw it looks like you've based this branch off main and not fsspec, so we are going to have diverging branches we will need to manage.. I think that hsould be ok as I don't think there will be too many conflicts, but before you merge any further PRs, let me merge in my other PR.. 

also make sure to do a squash to keep the history clean

**Comment by Karl5766** [comment]
Alright added back resources files and squashed the commits.

**Pull request merged**